src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMapping.cs

```diff
@@ -20,25 +22,30 @@ public class PackageSourceMapping
         /// </summary>
         internal IReadOnlyDictionary<string, IReadOnlyList<string>> Patterns { get; }
 
-        private Lazy<SearchTree> SearchTree { get; }
+        private Lazy<SearchTree?> SearchTree { get; }
 
         /// <summary>
         /// Indicate if any packageSource exist in package source mapping section
         /// </summary>
         public bool IsEnabled { get; }
 
         /// <summary>
-        /// Get package source names with matching prefix "packageId" from package source mapping section.
+        /// Get package source names with matching prefix "packageId" from package source mapping section, or an empty list if none are found.
         /// </summary>
         /// <param name="packageId">Search packageId. Cannot be null, empty, or whitespace only. </param>
-        /// <returns>Package source names with matching prefix "packageId" from package patterns.</returns>
+        /// <returns>Package source names with matching prefix "packageId" from package patterns, or an empty list if none are found.</returns>
         /// <exception cref="ArgumentException"> if <paramref name="packageId"/> is null, empty, or whitespace only.</exception>
         public IReadOnlyList<string> GetConfiguredPackageSources(string packageId)
         {
-            return SearchTree.Value?.GetConfiguredPackageSources(packageId);
+            if (SearchTree.Value is null)
+            {
+                return Array.Empty<string>();
+            }
+
+            return SearchTree.Value.GetConfiguredPackageSources(packageId);
```

Comment:
The SearchTree method can return null as well.

Reply:
The `SearchTree` returns an empty list
https://github.com/NuGet/NuGet.Client/blob/1fbe83fe674911a7f1289dad525be6ff73b7d2e5/src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/SearchTree.cs#L100